skymodel.py

```
@@ -0,0 +1,202 @@
+        options: ['none','Gaussian','Boxcar'] (default is 'none')
+    lsf_gauss_fwhm: float
+        Gaussian full-width half-max for line spread function wavelength bins.
+        Range > 0.0 (default is 5.0)
+    lsf_boxcar_fwhm: float
+        Boxcar full-width half-max for line spread function wavelength bins.
+        Range > 0.0 (default is 5.0)
+
+    observatory: str
+        Observatory where observation takes place.
+        Options are 'paranal', 'lasilla', 'armazones' (default is 'paranal')
+
+    Returns
+    -------
+    trans_waves, transmission: tuple of arrays of floats
+        'trans_waves' is an array of wavelengths in angstroms (float),
+        'transmission' is an array of fractional atmospheric
+        transmittance (float).
+
+    """
+
+    params = locals()
+
+    if params['observatory'] == 'lasilla':
+        params['observatory'] = '2400'
+    elif params['observatory'] == 'paranal':
+        params['observatory'] = '2640'
+    elif (params['observatory'] == '3060m' or
+          params['observatory'] == 'armazones'):
+        params['observatory'] = '3060'
+    else:
+        raise ValueError('Wrong Observatory name, please refer to the '
+                         'skycalc_cli documentation.')
+
+    # Use the bit from skycalc_cli which queries from the SkyCalc Sky Model
+    server = 'http://etimecalret-001.eso.org'
+    url = server + '/observing/etc/api/skycalc'
+    response = requests.post(url, data=json.dumps(params))
+    results = json.loads(response.text)
+
+    status = results['status']
+    tmpdir = results['tmpdir']
+    tmpurl = server + '/observing/etc/tmp/' + tmpdir + '/skytable.fits'
+
+    if status == 'success':
+        try:
+            response = requests.get(tmpurl, stream=True)
+            data = response.content
+        except requests.exceptions.RequestException as e:
+            print(e, 'could not retrieve FITS data from server')
+    else:
+        print('HTML request failed', results)
```

Should this raise an error if the HTML request fails, rather than printing "fail"?

---

That was my thinking too, I just wasn't sure what kind of error to use. I guess RuntimeError is for uncategorized errors?

---

You can pick one from https://docs.python.org/3/library/exceptions.html#exception-hierarchy or https://docs.python.org/3/library/urllib.error.html . You can also create a custom exception class.

---

I don't have much experience writing exception classes but I gave it a shot - any suggestions?